expertsystem/data.py

```diff
@@ -84,29 +99,121 @@ def __hash__(self) -> int:
         return hash(repr(self))
 
 
-class Particle(NamedTuple):
-    """Immutable data container for particle info."""
+@dataclass(frozen=True)
+class HasQuantumNumbers:  # pylint: disable=too-many-instance-attributes
```

Comment:
I'm personally don't like the class names with the `Has` prefix. To me the name suggests like a boolean.

Reply:
I know, but I wanted to signal usage somehow and couldn't use `_` here, see https://github.com/ComPWA/expertsystem/pull/135#issuecomment-668142352

Reply:
Note that these classes are really only to be used in the local scope, so I don't think naming is that crucial here. Also, PascalCase signals this is a class and cannot be a `bool`.